### DISCUSSION TITLE
Problem during setup with node version

### DISCUSSION THREAD

**Discussion by Searchingdays** [learningequality/kolibri#12276] [Q&A]
I was trying to setup kolibri for development, but am having problems with yarn run devservers and the node version. Moreover I am not able to use npm as it shows module not found error. When I try to use yarn to install npm, yarn shows error.

I have checked with node --version, inside the virtual environment, it is 18.19.0, but still yarn displays the error message  kolibri-tools@0.16.1-dev.1: The engine "node" is incompatible with this module. Expected version "18.x". Got "20.12.2"

**Comment by LianaHarris360** [comment]
Hi @Searchingdays,

Thanks for your interest in contributing!

I wonder if the node version 20.12.2 is the global version on your system and is being used rather than the node version listed in the virtual environment. Could you try switching the node version in your system terminal to v18?

The [npm documentation](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) recommends using a node version manager like [nvm](https://github.com/nvm-sh/nvm) to install Node.js and npm. In the Kolibri docs, [nodeenv](https://kolibri-dev.readthedocs.io/en/develop/howtos/nodeenv.html) is recommended, and I've had success using [volta](https://docs.volta.sh/guide/).

**Comment by Searchingdays** [comment]
I uninstalled the global node,js version and reinstalled it. Using nvm I installed the version 18.19.0, but then on yarn run devserver this was the error: kolibri-root@0.12.0-dev.1: The engine "node" is incompatible with this module. Expected version "20.x". Got "18.19.0".
I changed the version to 20.14.0 using nvm and this was the message:  kolibri-tools@0.16.1-dev.1: The engine "node" is incompatible with this module. Expected version "18.x". Got "20.14.0".    Totally confused.   I have been working inside Python virtual environment.

**Reply by MisRob** [reply]
@Searchingdays Can you try to use volta as @LianaHarris360 suggested? It has helped to many people with dependencies issues.

**Reply by MisRob** [reply]
@Searchingdays Have you made some local changes to `package.json` files?

**Comment by MisRob** [comment]
Doesn't seem to be active anymore, so closing
